### PR TITLE
feat: add TransactionsHandle function

### PR DESCRIPTION
**Pull request by smatthewenglish**
Initial attempt/sketch at a solution to [Issue #5143](https://github.com/paradigmxyz/reth/issues/5143)

### CODE REVIEW

File: crates/net/network/src/transactions.rs [hunk 1]
```diff
@@ -1204,6 +1222,12 @@ enum TransactionsCommand {
         peers: Vec<PeerId>,
         tx: oneshot::Sender<HashMap<PeerId, HashSet<TxHash>>>,
     },
+    /// Request specific transactions from a peer.
+    TransactionsHandle {
```
Comment (mattsse) [inline crates/net/network/src/transactions.rs:1226]:
```suggestion
    GetPeerSender {
```

File: crates/net/network/src/transactions.rs [hunk 2]
```diff
@@ -1204,6 +1222,12 @@ enum TransactionsCommand {
...
+        peer_id: PeerId,
+        requested_hashes: Vec<TxHash>,
+        result: oneshot::Sender<Result<RequestResult<PooledTransactions>, RecvError>>,
```
Comment (mattsse) [inline crates/net/network/src/transactions.rs:1229]: it's easier to return a clone of the Sender if the peer exists
```suggestion
        result: oneshot::Sender<Option<PeerRequestSender>>,
```

File: crates/net/network/src/transactions.rs [hunk 3]
```diff
@@ -1204,6 +1222,12 @@ enum TransactionsCommand {
         peers: Vec<PeerId>,
         tx: oneshot::Sender<HashMap<PeerId, HashSet<TxHash>>>,
     },
+    /// Request specific transactions from a peer.
+    TransactionsHandle {
+        peer_id: PeerId,
+        requested_hashes: Vec<TxHash>,
```
Comment (mattsse) [inline crates/net/network/src/transactions.rs:1228]:
```suggestion
```

File: crates/net/network/src/transactions.rs [hunk 4]
```diff
@@ -590,6 +590,24 @@ where
...
+                        Ok(transactions) => {
+                            let _ = result.send(Ok(RequestResult::Fulfilled(transactions)));
+                        }
+                        Err(e) => {
+                            let _ = result.send(Err(e));
+                        }
+                    }
+                } else {
+                    let _ = result.send(Err(RecvError::new("Peer not found")));
+                }
```
Comment (mattsse) [inline crates/net/network/src/transactions.rs:609]: this should return `peer.request_tx.clone()`

and all the request logic should happen in the context of the `TransactionHandle` where we can use await

**Review by mattsse — CHANGES_REQUESTED** [review]
great initial draft.

we can convert this into general purpose `GetPeerSender` command and the request handling can happen entirely outside of the TransactionManager in the Handle

File: crates/net/network/src/transactions.rs [hunk 5]
```diff
@@ -1204,6 +1222,11 @@ enum TransactionsCommand {
         peers: Vec<PeerId>,
         tx: oneshot::Sender<HashMap<PeerId, HashSet<TxHash>>>,
     },
+    /// Request specific transactions from a peer.
+    GetPeerSender {
+        peer_id: PeerId,
```
Comment (smatthewenglish) [inline crates/net/network/src/transactions.rs:1227]: that makes me think maybe we should remove `peer_id` 'cause it already exists on `PeerRequestSender`

**Review by mattsse — APPROVED** [review]

**Pull request merged**